galley/pkg/server/server.go

```diff
@@ -233,6 +231,15 @@ func newServer(a *Args, p patchTable) (*Server, error) {
 func getSourceSchema(a *Args) *schema.Instance {
 	b := schema.NewBuilder()
 	for _, spec := range kubeMeta.Types.All() {
+
+		if a.EnableServiceDiscovery {
```

Comment:
I don't think this is right. If the built-in types are enabled, we'd always want to use the specialized sources.

Reply:
I don't know what you mean by that? getSourceSchema is for signalling which resources to listen to the source. If Service discovery is enabled, we want to listen to all needed resources for that, regardless of whether the kind is excluded or not. IsBuiltIn is a proxy call for "is this type needed for service discovery".

Kind exclusion will control the serving of the resources. If you enable the service synthesis, then the source will still listen to the needed resources, but will not serve them through MCP (based on exclusion). The getMCPTypes method below controls which resources should be served through MCP.